lib/SimpleSAML/Kernel.php

```diff
@@ -1,170 +1,69 @@
-use Symfony\Component\DependencyInjection\Definition;
-use Symfony\Component\DependencyInjection\Loader\DirectoryLoader;
 use Symfony\Component\HttpKernel\Kernel as BaseKernel;
 use Symfony\Component\Routing\RouteCollectionBuilder;
 
-/**
- * A class to create the container and handle a given request.
- */
 class Kernel extends BaseKernel
 {
     use MicroKernelTrait;
 
-    const CONFIG_EXTS = '.{php,xml,yaml,yml}';
+    private const CONFIG_EXTS = '.{php,xml,yaml,yml}';
 
-    /** @var string */
-    private $module;
-
-
-    /**
-     * @param string $module
-     */
-    public function __construct(string $module)
-    {
-        $this->module = $module;
-
-        $env = getenv('APP_ENV') ?: (getenv('SYMFONY_ENV') ?: 'prod');
-
-        parent::__construct($env, false);
-    }
-
-
-    /**
-     * @return string
-     */
-    public function getCacheDir(): string
-    {
-        $configuration = Configuration::getInstance();
-        $cachePath = $configuration->getString('tempdir') . '/cache/' . $this->module;
-
-        if (0 === strpos($cachePath, '/')) {
-            return $cachePath;
-        }
-
-        return $configuration->getBaseDir() . '/' . $cachePath;
-    }
-
-
-    /**
-     * @return string
-     */
-    public function getLogDir(): string
+    public function registerBundles()
     {
-        $configuration = Configuration::getInstance();
-        $loggingPath = $configuration->getString('loggingdir');
-
-        if (0 === strpos($loggingPath, '/')) {
-            return $loggingPath;
+        $contents = require $this->getProjectDir().'/config/modules.php';
+        foreach ($contents as $class => $envs) {
+            if ($envs[$this->environment] ?? $envs['all'] ?? false) {
+                yield new $class();
+            }
         }
-
-        return $configuration->getBaseDir() . '/' . $loggingPath;
     }
 
-
-    /**
-     * @return array
-     */
-    public function registerBundles(): array
-    {
-        return [
-            new FrameworkBundle(),
-        ];
-    }
-
-
-    /**
-     * Get the module loaded in this kernel.
-     *
-     * @return string
-     */
-    public function getModule(): string
-    {
-        return $this->module;
-    }
-
-
-    /**
-     * Configures the container.
-     *
-     * @param ContainerBuilder $container
-     * @param LoaderInterface $loader
-     * @return void
-     */
-    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    protected function configureRoutes(RouteCollectionBuilder $routes)
     {
-        $configuration = Configuration::getInstance();
-        $baseDir = $configuration->getBaseDir();
-        $loader->load($baseDir . '/routing/services/*' . self::CONFIG_EXTS, 'glob');
-        $confDir = Module::getModuleDir($this->module) . '/routing/services';
-        if (is_dir($confDir)) {
-            $loader->load($confDir . '/**/*' . self::CONFIG_EXTS, 'glob');
+        foreach ($this->bundles as $bundle) {
```

Comment:
This iterator search in all bundles the ones that are SSP Modules and checks if they have a routes directory (and load the route files). The module name is used a prefix to the url.